static/js/modules/typeahead.js

```diff
@@ -0,0 +1,152 @@
+'use strict';
+var Handlebars = require('handlebars');
+var events = require('./events.js');
+
+module.exports = {
+  init: function() {
+    function filterCandidates(result) {
+      // Label House, President, Senate
+      var officeFull;
+      switch(result.elections[0].office_sought) {
+        case "H":
+          officeFull = "House of Representatives";
+          break;
+        case "S":
+          officeFull = "Senate";
+          break;
+        case "P":
+          officeFull = "President";
+          break;
+      }
+      var filteredResults =   {
```

Comment:
Its considered best practice to declare all `var` statements at the top of a function. You can populate them later. For example, you could do one of two things here:

```
function filterCandidates(result) {
  var officeFull,
        filteredResults;

  // stuff

  filteredResults = {
    // stuff
  }
}
```
or
```
function filterCandidates(result) {
  var officeFull,
        filteredResults = {};

  // stuff

  filteredResults.name = result.name;
  // and so on
}
```